apiserver/pkg/http/client.go

```diff
@@ -645,8 +646,8 @@ func (krc *KuberayAPIServerClient) extractStatus(bodyBytes []byte) (*rpcStatus.S
 	return status, nil
 }
 
-func (krc *KuberayAPIServerClient) createHttpRequest(method string, endPoint string, body io.Reader) (*http.Request, error) {
-	req, err := http.NewRequest(method, endPoint, body)
+func (krc *KuberayAPIServerClient) createHttpRequest(ctx context.Context, method string, endPoint string, body io.Reader) (*http.Request, error) {
```

Comment:
Is this a linter change? What's the difference between the old one (without ctx) and the new one (with ctx)?

Reply:
Yes, the linter error looks like follows:
```sh
pkg/http/client.go:654:29: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
        req, err := http.NewRequest(method, endPoint, body)
                                   ^
pkg/http/client.go:378:43: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
        httpRequest, err := krc.createHttpRequest("POST", createURL, bytes.NewReader(bytez))
                                                 ^
```

Reply:
The general practice for all IO operations is to add a context for timeout and cancellation.